src/test/java/teammates/test/cases/datatransfer/FeedbackQuestionAttributesTest.java

```diff
@@ -37,6 +40,230 @@ void setUpdatedAt(Instant updatedAt) {
 
     }
 
+    @Test
+    public void testValueOf() {
+        /* Solves the problem of "java.lang.NullPointerException: No API environment is registered for this thread on
+         Unit testing".
+         https://stackoverflow.com/questions/31994264/java-lang-nullpointerexception-no-api-environment-
+         is-registered-for-this-thread */
+        LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
```

Comment:
Attributes Unit Test should not touch things like `GaeSimulation`. BTW, we do we need this service?

Reply:
Without setting up the environment (and the id), an NPE would be encountered at the following line when `valueof()` is called:
https://github.com/TEAMMATES/teammates/blob/cbbfc98fee6210c711e1cefe31afe623640e3702/src/main/java/teammates/storage/entity/FeedbackQuestion.java#L127-L129

This is due to 
https://github.com/TEAMMATES/teammates/blob/f49c5ba276787a93287a2bb11e800e1a780a85fb/src/main/java/teammates/storage/entity/FeedbackQuestion.java#L90

Reply:
Does the suggestion in https://github.com/TEAMMATES/teammates/pull/8739#issuecomment-378828833 work?

Reply:
Tried it, couldn't get it to work. The issue is the `feedbackQuestionId` is still null. `db.saveEntity()` will make a call `db.makeAttributes()`, which then makes a call to `attribute.valueOf()`, and we come back the method that we're trying to test in the first place. That also means the original issue is not being solved because the NPE will be thrown again.

Reply:
Try making this class extend from `BaseAttributesTest` - a minimal GAE environment is set up there.